d3-series/materials/Week 3/Assignment/Viz3.html

```diff
@@ -0,0 +1,85 @@
+<html>
+
+    <head>
+        <title>TEST</title>
+    </head>
+
+    <body>
+        <div id="viz"></div>
+
+        <script src="https://d3js.org/d3.v4.min.js"></script>
+        <script>
+    
+            var height = 800;
+            var width = 1000;
+
+            var svg = d3.select("#viz")
+                .append('svg')
+                .attr('height', height)
+                .attr('width', width);
+
+            d3.json('data.json', function (data) {
+
+                updateViz(data);
+
+                var int = setInterval(function () {
+                    data.pop();
+                    updateViz(data);
+
+                    if (data.length === 0) {
+                        clearInterval(int);
+                    }
+                }, 2000);
+                });
+
+            function updateViz(data) {
+                var cx = 50;
+                var cy = 50;
+                var cir = svg.selectAll('circle')
+                    .data(data)
+                
+                cir.enter()
+                    .append('circle')
+                    .attr('cx',function(d,i){
+                        return Math.random()*1000;
+                        
+                    })
+                    .attr('cy',function(d){
+                        return Math.random()*800;
+                        ;
+                    })
+                    .attr('r',function(d){
+                        return d.size;
+                    })
+                    .attr('fill', function (d) {
+                        return d.color;});
+                cir.exit()
+                   .remove();
+                
+                cir.attr('cx',function(d,i){

```

Comment:
There are two ways to approach updating the data, one is to remove each element after each 2 sec period but with the same positioning and the other is to update the positioning while also removing an element after each 2 sec period. Since you have an update pattern as well as the enter/exit, the positioning will be updated after every 2 seconds, but if you only had the enter/exit, the circles would remain in the same location but would disappear one by one. No specific feedback but just as an fyi!